### PR TITLE
P3222R0 Fix C++26 by adding transposed special cases for P2642 layouts

### LINKED ISSUE #7421: [2024-11 LWG Motion 9] P3222R0 Fix C++26 by adding transposed special cases for P2642 layouts
[P3222R0](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2024/p3222r0.html) (Fix C++26 by adding transposed special cases for P2642 layouts

### PR DESCRIPTION
**Pull request by tkoeppe**
Fixes https://github.com/cplusplus/papers/issues/1870.
Fixes #7421.

### CODE REVIEW

**Review by jwakely — APPROVED** [review]

File: source/numerics.tex [hunk 1]
```diff
@@ -13108,11 +13116,26 @@
...
+R(a.data_handle(), ReturnMapping(@\exposid{transpose-extents}@(a.mapping().extents()),
+  a.mapping().stride(1)), a.accessor())
+\end{codeblock}
+if \tcode{Layout} is \tcode{layout_left_padded<PaddingValue>}
+for some \tcode{size_t} value \tcode{PaddingValue};
+\item
+otherwise,
+\begin{codeblock}
+R(a.data_handle(), ReturnMapping(@\exposid{transpose-extents}@(a.mapping().extents()),
+  a.stride(0)), a.accessor())
```
Comment (jwakely) [inline source/numerics.tex:13130]: Actually, shouldn't this be:

```suggestion
  a.mapping().stride(0)), a.accessor())
```

@mhoemmen

Comment (jwakely) [reply source/numerics.tex:13130]: If we can fix this editorially now that would save going through the issue process. There is no `stride` member of `mdspan` so the approved wording is nonsensical.

Comment (tkoeppe) [reply source/numerics.tex:13130]: I see now, yes, makes sense. Will do!

**Review by jwakely — APPROVED** [review]
I'm confident the editorial fix is correct.

I hope @mhoemmen will confirm.

File: source/numerics.tex (hunk 1 above)
Comment (mhoemmen) [reply source/numerics.tex]: @jwakely @tkoeppe Hi! : - )  Good catch!  Somehow I missed that one for the `layout_right_padded` case, even though I got it right for the `layout_left_padded` case right above it.  `a.mapping().stride(0)` is absolutely what I intended to write there.

Comment (jwakely) [reply source/numerics.tex]: Thanks!

**Comment by mhoemmen** [comment]
@jwakely I have reviewed the changes and I confirm the fix.  Thank you! : - )

**Pull request merged**